src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
+    final Collection<Unit> airUnits = Match.getMatches(units, ProMatches.unitIsAlliedAir(player, data));
+    List<Unit> airThatCantLand = new ArrayList<Unit>();
+    for (final Unit airUnit : airUnits) {
+      final UnitAttachment ua = UnitAttachment.get(airUnit.getType());
+      final int cost = ua.getCarrierCost();
+      if (cost != -1) {
```

Comment:
agree, my last comment is more or less on the same lines. I think the unnecessary "public" vs "private" method is a liability. I'd rather you start to move on though, this has been reviewed pretty closely now IMO :)